app/src/main/java/com/github/multimatum_team/multimatum/activity/GroupInviteActivity.kt

```diff
@@ -0,0 +1,123 @@
+package com.github.multimatum_team.multimatum.activity
+
+import android.os.Bundle
+import android.view.View
+import android.widget.Button
+import android.widget.TextView
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import com.github.multimatum_team.multimatum.LogUtil
+import com.github.multimatum_team.multimatum.R
+import com.github.multimatum_team.multimatum.model.GroupID
+import com.github.multimatum_team.multimatum.model.SignedInUser
+import com.github.multimatum_team.multimatum.model.UserGroup
+import com.github.multimatum_team.multimatum.viewmodel.AuthViewModel
+import com.github.multimatum_team.multimatum.viewmodel.GroupViewModel
+import com.google.firebase.dynamiclinks.FirebaseDynamicLinks
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+
+/**
+ *  Activity to accept or deny group invites.
+ */
+@AndroidEntryPoint
+class GroupInviteActivity : AppCompatActivity() {
+    @Inject
+    lateinit var dynamicLinks: FirebaseDynamicLinks
+
+    private val authViewModel: AuthViewModel by viewModels()
+    private val groupViewModel: GroupViewModel by viewModels()
+
+    private lateinit var inviteMessageView: TextView
+    private lateinit var groupNameView: TextView
+    private lateinit var acceptButton: Button
+    private lateinit var denyButton: Button
+
+    private lateinit var currentUser: SignedInUser
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_group_invite)
+
+        LogUtil.debugLog(intent.toString())
+
+        inviteMessageView = findViewById(R.id.group_invite_message)
+        groupNameView = findViewById(R.id.group_invite_name)
+        acceptButton = findViewById(R.id.group_invite_accept)
+        denyButton = findViewById(R.id.group_invite_deny)
+
+        groupNameView.visibility = View.INVISIBLE
+        acceptButton.visibility = View.INVISIBLE
+        denyButton.visibility = View.INVISIBLE
+
+        initView()
+    }
+
+    private fun initView() {
+        when (val user = authViewModel.getUser().value) {
+            is SignedInUser -> {
+                currentUser = user
+                dynamicLinks
+                    .getDynamicLink(intent)
+                    .addOnSuccessListener(this) { pendingDynamicLinkData ->
+                        when (val link = pendingDynamicLinkData?.link) {
+                            null -> initInvalidLink()
+                            else -> {
+                                when (val groupID = link.getQueryParameter("id")) {
+                                    null -> initInvalidLink()
+                                    else -> initValidLink(groupID)
+                                }
+                            }
+                        }
+                    }
+                    .addOnFailureListener(this) { e ->
+                        initInvalidLink()
+                    }
```

Comment:
Maybe you can extract function(s) in this code block? This would make the code more readable and the name of the function(s) could help understanding what the purpose of this code block is.

Reply:
Sure, I extracted the code and added comments to explain the logic